#BluetoothMesh demo for Generic OnOff & Level Models

vikrant8052 opened the pull request:
This is a simple application demonstrating a Bluetooth Mesh Node in 
which Root element has following models
- Generic OnOff Server
- Generic OnOff Client
- Generic Level Server
- Generic Level Client 

And Secondary element has following models
- Generic OnOff Server
- Generic OnOff Client

Signed-off-by: vikrant p. more   Email: vikrant8051@gmail.com

vikrant8052 committed:
mesh_demo

vikrant8052 committed:
mesh_demo

vikrant8052 committed:
removed some lines & spaces

vikrant8052 committed:
bug fix

dbkinder commented on samples/boards/nrf52/mesh/onoff_level_app/README.rst:
```diff
@@ -0,0 +1,132 @@
+Overview
+********
+
+This is a simple application demonstrating a Bluetooth mesh node.
```
It would be better not to have so many blank lines (for readability):

    This is a simple application demonstrating a Bluetooth mesh node.
    The node's root element has the following models:

    - Generic OnOff Server
    - Generic OnOff Client
    - Generic Level Server
    - Generic Level Client 

    The node's secondary element has the following models:

    - Generic OnOff Server
    - Generic OnOff Client

dbkinder commented on samples/boards/nrf52/mesh/onoff_level_app/README.rst:
```diff
@@ -0,0 +1,132 @@
+  * [Button3: publishes Level = 25%]
+  * [Button4: publishes Level = 100%]
+
+-------------------------------------------------------------------------------------------------------------------------
```
delete this line of dashes

dbkinder commented on samples/boards/nrf52/mesh/onoff_level_app/README.rst:
```diff
@@ -0,0 +1,132 @@
+
+After provisioning, the button clients must
+be configured to publish and the LED servers to subscribe.
+
```
delete this blank line (no need to start a new paragraph).

dbkinder commented on samples/boards/nrf52/mesh/onoff_level_app/README.rst:
```diff
@@ -0,0 +1,132 @@
+   :compact:
+
+Provisioning is done using the BlueZ meshctl utility. 
+
```
delete this blank line

dbkinder commented on samples/boards/nrf52/mesh/onoff_level_app/README.rst:
```diff
@@ -0,0 +1,132 @@
+
+Provisioning is done using the BlueZ meshctl utility. 
+
+Here is an example that binds 
```
change to:

    In this example, we'll use meshctl commands to bind:

dbkinder commented on samples/boards/nrf52/mesh/onoff_level_app/README.rst:
```diff
@@ -0,0 +1,132 @@
+supported by the node. This example supports 2 elements per node.
+
+The first or root element of the node contains models for configuration,
+health, and above mentioned models. The secondary elements only
```
and the above

dbkinder commented:
There are lots of sanitycheck errors noted in the [shippable log details](https://app.shippable.com/github/zephyrproject-rtos/zephyr/runs/16467/summary)

vikrant8051 commented:
Hi @dbkinder,

How I could check that sanitycheck errors or zephyr code convention error ?
Tommorow I will make suggested changes.

Could you please meanwhile refer this PR to someone, to check out programming Bugs ?

Hi @jhedberg, could you please help me to find out implementation or programming level bugs  ?

vikrant8052 committed:
update README.rst

vikrant8052 committed:
Update README.rst

vikrant8052 committed:
Update README.rst

vikrant8052 committed:
Update README.rst

vikrant8052 committed:
Update README.rst

vikrant8051 commented:
@dbkinder I've make suggested changes.

vikrant8052 committed:
minor improvements

vikrant8052 committed:
minor improvements for NVS

vikrant8052 committed:
minor improvements for NVS

vikrant8052 committed:
minor improvements for NVS

vikrant8052 committed:
minor update

vikrant8052 committed:
TID bug solved

vikrant8052 committed:
TID bug solved

vikrant8052 committed:
pulish_++tid_to_tid++

vikrant8052 committed:
removed_tid_concept

vikrant8052 committed:
Update DEVICE_composition.c

vikrant8052 committed:
Update DEVICE_composition.h

vikrant8052 committed:
update doer.c

vikrant8052 committed:
implemented opcode function of GEN_DELTA_SRV_UNACK

vikrant8052 committed:
update doer.c

vikrant8052 committed:
doer.c

vikrant8052 committed:
doer.c

vikrant8051 commented:
@dbkinder is anything pending from my side ?

dbkinder commented on samples/boards/nrf52/mesh/onoff_level_app/README.rst:
```diff
@@ -0,0 +1,106 @@
+.. _bluetooth-mesh-onoff-level-sample:
+Bluetooth: Mesh OnOff & Level Model
```
Got rid of too many blank lines :)  
Put a blank line before this document title.

dbkinder commented on samples/boards/nrf52/mesh/onoff_level_app/README.rst:
```diff
@@ -0,0 +1,106 @@
+- Generic OnOff Client
+- Generic Level Server
+- Generic Level Client 
+And Secondary element has following models
```
Needs a blank line after a list, so add a blank line before this.

dbkinder commented on samples/boards/nrf52/mesh/onoff_level_app/README.rst:
```diff
@@ -0,0 +1,106 @@
+
+- Generic OnOff Server
+- Generic OnOff Client
+Prior to provisioning, an unprovisioned beacon is broadcast that contains
```
Needs a blank line after a list, so add a blank line before this.

dbkinder commented on samples/boards/nrf52/mesh/onoff_level_app/README.rst:
```diff
@@ -0,0 +1,106 @@
+
+  * [Button3: publishes Level = 25%]
+  * [Button4: publishes Level = 100%]
+After provisioning, the button clients must
```
Needs a blank line after a list, so add a blank line before this.

dbkinder commented on samples/boards/nrf52/mesh/onoff_level_app/README.rst:
```diff
@@ -0,0 +1,106 @@
+   :board: nrf52840_pca10056
+   :goals: build flash
+   :compact:
+Provisioning is done using the BlueZ meshctl utility. In this example, we'll use meshctl commands to bind:  
```
Needs a blank line after a directive's content block, so add a blank line before this.

dbkinder commented on samples/boards/nrf52/mesh/onoff_level_app/README.rst:
```diff
@@ -0,0 +1,106 @@
+  to publish to group 0xC000 and LED1 to subscribe to that group.
+- Button3, Button4, and LED3 to application key 1. It then configures Button3 and Button4
+  to publish to group 0xC000 and LED3 to subscribe to that group.
+.. code-block:: console
```
Needs a blank line after a list, so add a blank line before this.

dbkinder commented on samples/boards/nrf52/mesh/onoff_level_app/README.rst:
```diff
@@ -0,0 +1,106 @@
+   sub-add 0100 c000 1002
+   pub-set 0100 c000 1 0 5 1001
+   pub-set 0100 c000 1 0 5 1003 
+The meshctl utility maintains a persistent JSON database containing
```
Needs a blank line after a code-block directive's content block, so add a blank line before this.

dbkinder reviewed (changes requested):
Got rid of a few too many blank lines :)

Take a look at the shippable log: https://app.shippable.com/github/zephyrproject-rtos/zephyr/runs/16621/5/tests

Hopefully one of the developers (I'm a tech writer) can help with these problems:  @jhedberg ?

There are many shippable complaints about missing signed-off-by lines, malformed commit titles, and missing commit message bodies (all commit messages must contain a long description and not just a title).

The checkpatch tool is also complaining about trailing white space, long lines, and coding convention errors.

vikrant8052 committed:
update README.rst

vikrant8052 committed:
removed unnecessary spaces & lines

vikrant8052 committed:
removed unnecessary spaces & lines

vikrant8052 committed:
This is a simple application demonstrating a Bluetooth Mesh Node in
which Root element has following models

    Generic OnOff Server
    Generic OnOff Client
    Generic Level Server
    Generic Level Client

And Secondary element has following models

    Generic OnOff Server
    Generic OnOff Client

Signed-off-by: vikrant p. more Email: vikrant8051@gmail.com

vikrant8052 committed:
This is a simple application demonstrating a Bluetooth Mesh Node in
which Root element has following models

    Generic OnOff Server
    Generic OnOff Client
    Generic Level Server
    Generic Level Client

And Secondary element has following models

    Generic OnOff Server
    Generic OnOff Client

Signed-off-by: vikrant p. more Email: vikrant8051@gmail.com

vikrant8052 committed:
This is a simple application demonstrating a Bluetooth Mesh Node in
which Root element has following models

- Generic OnOff Server
- Generic OnOff Client
- Generic Level Server
- Generic Level Client

And Secondary element has following models

- Generic OnOff Server
- Generic OnOff Client

Signed-off-by: vikrant p. more Email: vikrant8051@gmail.com

vikrant8052 committed:
This is a simple application demonstrating a Bluetooth mesh node in
which Root element has following models

- Generic OnOff Server
- Generic OnOff Client
- Generic Level Server
- Generic Level Client

And Secondary element has following models

- Generic OnOff Server
- Generic OnOff Client

Signed-off-by: vikrant p. more Email: vikrant8051@gmail.com
